app/validation.py

```diff
@@ -68,22 +80,25 @@ def validate_user_auth_json_or_400(submitted_json):
         abort(400, "JSON was not a valid format. {}".format(e.message))
 
 
-def validate_json_or_400(submitted_json):
-    if not validate_json(submitted_json):
+def detect_framework_or_400(submitted_json):
+    framework = detect_framework(submitted_json)
+    if not framework:
         abort(400, "JSON was not a valid format. {}".format(
             reason_for_failure(submitted_json))
         )
-
-
-def validate_json(submitted_json):
-    if validates_against_schema(G6_SCS_VALIDATOR, submitted_json):
-        return 'G6-SCS'
-    elif validates_against_schema(G6_SAAS_VALIDATOR, submitted_json):
-        return 'G6-SaaS'
-    elif validates_against_schema(G6_PAAS_VALIDATOR, submitted_json):
-        return 'G6-PaaS'
-    elif validates_against_schema(G6_IAAS_VALIDATOR, submitted_json):
-        return 'G6-IaaS'
+    return framework
+
+
+def detect_framework(submitted_json):
```

Comment:
I find this a confusing name. While it does detect the framework this function is mostly used for validating service JSON.